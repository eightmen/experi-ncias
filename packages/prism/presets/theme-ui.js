// @ts-check

const syntaxStyles = {
  '.comment': {
    fontStyle: 'italic',
  },
  '.property, .tag, .boolean, .number, .constant, .symbol, .function, .class-name, .regex, .important, .variable':
    { color: 'accent' },
  '.atrule, .attr-value, .keyword': {
    color: 'primary',
  },
  '.selector, .attr-name, .string, .char, .builtin, .inserted': {
    color: 'secondary',
  },
  '.deleted': {
    color: 'accent',
    fontStyle: 'italic',
  },
  '.comment,.prolog,.doctype,.cdata,.punctuation.punctuation,.operator,.entity,.url':
    { color: 'gray' },
}

export default syntaxStyles
