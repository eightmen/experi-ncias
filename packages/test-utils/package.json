
{
  "name": "@theme-ui/test-utils",
  "version": "0.15.6-develop.0",
  "private": true,
  "license": "MIT",
  "repository": "system-ui/theme-ui",
  "source": "src/index.ts",
  "main": "dist/theme-ui-test-utils.cjs.js",
  "module": "dist/theme-ui-test-utils.esm.js",
  "types": "dist/theme-ui-test-utils.cjs.d.ts",
  "sideEffects": false,
  "scripts": {
    "build": "tsc",
    "watch": "tsc -w",
    "typecheck": "tsc --noEmit"
  },
  "dependencies": {
    "@testing-library/react": "^13.3.0",
    "conditional-type-checks": "^1.0.6",
    "react-test-renderer": "^18.2.0",
    "ts-snippet": "^5.0.0"
  },
  "peerDependencies": {
    "react": ">=18",
    "theme-ui": "workspace:^"
  },
  "devDependencies": {
    "@types/react": "^18",
    "react": "^18",
    "react-dom": ">=18",
    "typescript": ">=4",
    "theme-ui": "workspace:^"
  }
}