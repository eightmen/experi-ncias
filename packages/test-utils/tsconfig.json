{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "target": "ES2018"
  },
  "include": ["src/**/*.ts", "src/**/*.tsx", "src/**/*.d.ts"]
}
