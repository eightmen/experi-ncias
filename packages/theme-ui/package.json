
{
  "name": "theme-ui",
  "version": "0.15.6-develop.0",
  "description": "The Design Graph Framework",
  "source": "src/index.ts",
  "main": "dist/theme-ui.cjs.js",
  "module": "dist/theme-ui.esm.js",
  "types": "dist/theme-ui.cjs.d.ts",
  "sideEffects": false,
  "author": "Brent Jackson <jxnblk@gmail.com>",
  "license": "MIT",
  "repository": "system-ui/theme-ui",
  "scripts": {},
  "dependencies": {
    "@theme-ui/color-modes": "workspace:^",
    "@theme-ui/components": "workspace:^",
    "@theme-ui/core": "workspace:^",
    "@theme-ui/css": "workspace:^",
    "@theme-ui/global": "workspace:^",
    "@theme-ui/theme-provider": "workspace:^"
  },
  "peerDependencies": {
    "react": ">=18"
  },
  "devDependencies": {
    "react": ">=18",
    "react-dom": ">=18",
    "@types/react": "^18",
    "@theme-ui/test-utils": "workspace:^"
  },
  "keywords": [
    "theme-ui",
    "emotion",
    "mdx",
    "css",
    "styles",
    "css-in-js"
  ],
  "preconstruct": {
    "entrypoints": [
      "index.ts",
      "jsx-dev-runtime.ts",
      "jsx-runtime.ts"
    ]
  }
}