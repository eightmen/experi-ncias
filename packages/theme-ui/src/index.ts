import { jsx as coreJsx } from '@theme-ui/core'
import type { ThemeUIJSX, ThemeUIStyleObject } from '@theme-ui/core'
import React from 'react'

export {
  __ThemeUIContext,
  merge,
  useThemeUI,
  createElement,
} from '@theme-ui/core'
export type {
  ThemeUIContextValue,
  SxProp,
  ColorMode,
  ColorModesScale,
  CSSObject,
  CSSProperties,
  CSSPseudoSelectorProps,
  ResponsiveStyleValue,
  ThemeUICSSProperties,
  ThemeUIStyleObject,
  ThemeUICSSObject,
  Theme,
  ThemeStyles,
  TLengthStyledSystem,
  StylePropertyValue,
} from '@theme-ui/core'
export { useColorMode, InitializeColorMode } from '@theme-ui/color-modes'
export { ThemeProvider } from '@theme-ui/theme-provider'
export { default as Global } from '@theme-ui/global'
export * from '@theme-ui/components'
export { css, get } from '@theme-ui/css'

export const BaseStyles = (
  props: Record<string, unknown> & { sx?: ThemeUIStyleObject }
): JSX.Element =>
  jsx('div', {
    ...props,
    sx: {
      fontFamily: 'body',
      lineHeight: 'body',
      fontWeight: 'body',
      variant: 'styles',
      ...props.sx,
    },
  })

export const jsx = coreJsx

export declare namespace jsx {
  export namespace JSX {
    export interface Element extends ThemeUIJSX.Element {}
    export interface ElementClass extends ThemeUIJSX.ElementClass {}
    export interface ElementAttributesProperty
      extends ThemeUIJSX.ElementAttributesProperty {}
    export interface ElementChildrenAttribute
      extends ThemeUIJSX.ElementChildrenAttribute {}
    export type LibraryManagedAttributes<C, P> =
      ThemeUIJSX.LibraryManagedAttributes<C, P>
    export interface IntrinsicAttributes
      extends ThemeUIJSX.IntrinsicAttributes {}
    export interface IntrinsicClassAttributes<T>
      extends ThemeUIJSX.IntrinsicClassAttributes<T> {}
    export interface IntrinsicElements extends ThemeUIJSX.IntrinsicElements {}
  }
}
