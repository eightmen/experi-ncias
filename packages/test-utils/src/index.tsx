import * as React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import * as tsSnippet from 'ts-snippet'
import { Theme, ThemeProvider } from 'theme-ui'

export * from '@testing-library/react'

export type {
  Assert,
  AssertFalse,
  AssertTrue,
  Has,
  IsAny,
  IsExact,
  IsNever,
  IsNullable,
  IsUnknown,
  NotHas,
} from 'conditional-type-checks'

export const renderHook = <T,>(
  useHook: () => T,
  options: { theme?: Theme } = {}
): {
  unmount: () => void
  rerender: (
    ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>
  ) => void
  readonly result: NonNullable<T>
} => {
  let value: T | undefined = undefined

  const Component = () => {
    value = useHook()
    return null
  }

  const { theme } = options
  const { unmount, rerender } = render(<Component />, {
    wrapper: theme && ((props) => <ThemeProvider theme={theme} {...props} />),
  })

  return {
    unmount,
    rerender,
    get result() {
      return value!
    },
  }
}

export const renderJSON = (
  el: Parameters<typeof renderer.create>[0]
): renderer.ReactTestRendererJSON | null => {
  const json = renderer.create(el).toJSON()
  if (Array.isArray(json)) {
    return json[0]
  }

  return json
}

/**
 * @param setupCode imports, common code
 * @example
 * const expectSnippet = expecter(`
 *  import { add } from './made-up/add'
 * `);
 *
 * expectSnippet(`
 *   const x = add(10, 20)
 *