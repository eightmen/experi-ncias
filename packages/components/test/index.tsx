/**
 * @jest-environment jsdom
 */

import React from 'react'
import { renderJSON } from '@theme-ui/test-utils'
import { ThemeProvider } from 'theme-ui'
import {
  Card,
  Flex,
  Heading,
  Image,
  Link,
  Paragraph,
  Text,
  Label,
  Input,
  Select,
  Textarea,
  Radio,
  Checkbox,
  Slider,
  Field,
  Progress,
  Donut,
  Spinner,
  Avatar,
  Badge,
  Close,
  Alert,
  Divider,
  Embed,
  AspectRatio,
  AspectImage,
  Container,
  NavLink,
  Message,
  IconButton,
  MenuButton,
  Switch,
} from '..'

import { theme } from './__test-utils__'

describe('Card', () => {
  test('renders', () => {
    const json = renderJSON(
      <ThemeProvider theme={theme}>
        <Card />
      </ThemeProvider>
    )
    expect(json).toMatchSnapshot()
  })
})

describe('Flex', () => {
  test('renders', () => {
    const json = renderJSON(
      <ThemeProvider theme={theme}>
        <Flex />
      </ThemeProvider>
    )
    expect(json).toMatchSnapshot()
  })
})

describe('Heading', () => {
  test('renders', () => {
    const json = renderJSON(
      <ThemeProvider theme={theme}>
        <Heading />
      </ThemeProvider>
    )
    expect(json).toMatchSnapshot()
  })
})

describe('Image', () => {
  test('renders', () => {
    const json = renderJSON(
      <ThemeProvider theme={theme}>
        <Image variant="avatar" />
      </ThemeProvider>
    )
    expect(json).toMatchSnapshot()
  })
})

describe('Link', () => {
  test('renders', () => {
    const json = renderJSON(
      <ThemeProvider theme={theme}>
        <Link variant="nav" />
      </ThemeProvider>
    )
    expect(json).toMatchSnapshot()
  })
})

describe('Paragraph', () => {
  test('renders', () => {
    const json = renderJSON(
      <ThemeProvider theme={theme}>
        <Paragraph />
      </ThemeProvider>
    )
    expect(json).toMatchSnapshot()
  })

  test('renders with variant prop', () => {
    const variant = 'block'
    const json = renderJSON(
      <ThemeProvider theme={theme}>
        <Paragraph varian