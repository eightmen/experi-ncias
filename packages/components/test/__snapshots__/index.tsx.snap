// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  font-weight: bold;
  color: white;
  background-color: primary;
  border-radius: 4px;
}

<div
  className="emotion-0"
/>
`;

exports[`AspectImage renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
  overflow: hidden;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  height: 0;
  padding-bottom: 75%;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  object-fit: cover;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  />
  <div
    className="emotion-2"
  >
    <img
      className="emotion-3"
    />
  </div>
</div>
`;

exports[`AspectRatio renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: relative;
  overflow: hidden;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  height: 0;
  padding-bottom: 75%;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
}

<div
  className="emotion-0"
>
  <div
    className="emotion-1"
  />
  <div
    className="emotion-2"
  />
</div>
`;

exports[`Avatar renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  border-radius: 9999px;
  width: 48px;
  height: 48px;
  border-radius: 99999px;
}

<img
  className="emotion-0"
  height={48}
  width={48}
/>
`;

exports[`Badge renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  vertical-align: baseline;
  font-size: 12px;
  font-weight: bold;
  white-space: nowrap;
  padding-left: 4px;
  padding-right: 4px;
  border-radius: 2px;
  color: white;
  background-color: primary;
}

<div
  className="emotion-0"
/>
`;

exports[`Card renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
  background-color: muted;
  border-radius: 4px;
}

<div
  className="emotion-0"
/>
`;

exports[`Checkbox renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  min-width: -webkit-min-content;
  min-width: -moz-min-content;
  min-width: min-content;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: absolute;
  opacity: 0;
  z-index: -1;
  width: 1px;
  height: 1px;
  overflow: hidden;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: none;
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: 8px;
  border-radius: 4px;
  color: gray;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

input:checked~.emotion-2 {
  display: block;
}

input:checked~.emotion-2 {
  color: primary;
}

input:focus~.emotion-2 {
  color: primary;
  background-color: highlight;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: 8px;
  border-radius: 4px;
  color: gray;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

input:checked~.emotion-3 {
  display: none;
}

input:checked~.emotion-3 {
  color: primary;
}

input:focus~.emotion-3 {
  color: primary;
  background-color: highlight;
}

<div
  className="emotion-0"
>
  <input
    className="emotion-1"
    type="checkbox"
  />
  <svg
    aria-hidden="true"
    className="emotion-2"
    fill="currentcolor"
    height={24}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
    />
  </svg>
  <svg
    aria-hidden="true"
    className="emotion-3"
    fill="currentcolor"
    height={24}
    viewBox="0 0 24 24"
    width={24}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
    />
  </svg>
</div>
`;

exports[`Close renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 4px;
  width: 32px;
  height: 32px;
  color: inherit;
  background-color: transparent;
  border: none;
  border-radius: 4px;
}

<button
  aria-label="Close"
  className="emotion-0"
  title="Close"
>
  <svg
    fill="currentColor"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
    />
  </svg>
</button>
`;

exports[`Container renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  max-width: container;
  margin-left: auto;
  margin-right: auto;
}

<div
  className="emotion-0"
/>
`;

exports[`Divider renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: gray;
  margin: 0;
  margin-top: 8px;
  margin-bottom: 8px;
  border: 0;
  border-bottom: 1px solid;
}

<hr
  className="emotion-0"
/>
`;

exports[`Donut renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: primary;
}

<svg
  aria-valuemax={1}
  aria-valuemin={0}
  aria-valuenow={0}
  className="emotion-0"
  fill="none"
  height={128}
  role="img"
  stroke="currentcolor"
  strokeWidth={2}
  viewBox="0 0 32 32"
  width={128}
>
  <title>
    Donut
  </title>
  <circle
    cx={16}
    cy={16}
    opacity={0.125}
    r={14}
  />
  <circle
    cx={16}
    cy={16}
    r={14}
    strokeDasharray={87.96459430051421}
    strokeDashoffset={87.96459430051421}
    transform="rotate(-90 16 16)"
  />
</svg>
`;

exports[`Embed renders with box system props 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  height: 0;
  padding-bottom: 56.25%;
  position: relative;
  overflow: hidden;
  margin-left: auto;
  margin-right: auto;
  padding-top: 32px;
  padding-bottom: 32px;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  bottom: 0;
  left: 0;
  border: 0;
}

<div
  className="emotion-0"
>
  <iframe
    allowFullScreen={true}
    className="emotion-1"
    frameBorder={0}
    height={315}
    width={560}
  />
</div>
`;

exports[`Embed renders with title 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  height: 0;
  padding-bottom: 56.25%;
  position: relative;
  overflow: hidden;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  bottom: 0;
  left: 0;
  border: 0;
}

<div
  className="emotion-0"
>
  <iframe
    allowFullScreen={true}
    className="emotion-1"
    frameBorder={0}
    height={315}
    title="Embed"
    width={560}
  />
</div>
`;

exports[`Field renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
}

.emotion-2:autofill,
.emotion-2:autofill:hover,
.emotion-2:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-2:autofill:first-line,
.emotion-2:autofill:hover:first-line,
.emotion-2:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-2:-webkit-autofill,
.emotion-2:-webkit-autofill:hover,
.emotion-2:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-2:-webkit-autofill:first-line,
.emotion-2:-webkit-autofill:hover:first-line,
.emotion-2:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

<div
  className="emotion-0"
>
  <label
    className="emotion-1"
  />
  <input
    className="emotion-2"
  />
</div>
`;

exports[`Field renders with id prop 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
}

.emotion-2:autofill,
.emotion-2:autofill:hover,
.emotion-2:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-2:autofill:first-line,
.emotion-2:autofill:hover:first-line,
.emotion-2:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-2:-webkit-autofill,
.emotion-2:-webkit-autofill:hover,
.emotion-2:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-2:-webkit-autofill:first-line,
.emotion-2:-webkit-autofill:hover:first-line,
.emotion-2:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

<div
  className="emotion-0"
>
  <label
    className="emotion-1"
    htmlFor="test-field"
  />
  <input
    className="emotion-2"
    id="test-field"
  />
</div>
`;

exports[`Flex renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="emotion-0"
/>
`;

exports[`Heading renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: heading;
  font-weight: heading;
  line-height: heading;
  font-size: 32px;
}

<h2
  className="emotion-0"
/>
`;

exports[`IconButton renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 4px;
  width: 32px;
  height: 32px;
  color: inherit;
  background-color: transparent;
  border: none;
  border-radius: 4px;
}

<button
  className="emotion-0"
/>
`;

exports[`Image renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  width: 48px;
  height: 48px;
  border-radius: 99999px;
}

<img
  className="emotion-0"
/>
`;

exports[`Input renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
}

.emotion-0:autofill,
.emotion-0:autofill:hover,
.emotion-0:autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-0:autofill:first-line,
.emotion-0:autofill:hover:first-line,
.emotion-0:autofill:focus:first-line {
  font-size: 1rem;
}

.emotion-0:-webkit-autofill,
.emotion-0:-webkit-autofill:hover,
.emotion-0:-webkit-autofill:focus {
  box-shadow: inset 0 0 0 1000px var(--theme-ui-input-autofill-bg);
  font-size: inherit;
}

.emotion-0:-webkit-autofill:first-line,
.emotion-0:-webkit-autofill:hover:first-line,
.emotion-0:-webkit-autofill:focus:first-line {
  font-size: 1rem;
}

<input
  className="emotion-0"
/>
`;

exports[`Label renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<label
  className="emotion-0"
/>
`;

exports[`Link renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-weight: normal;
}

<a
  className="emotion-0"
/>
`;

exports[`MenuButton renders 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inli